scripts/increment_auto_off.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+"""A program that ticks a counter down until deactivating log_persistence"""
+
+import logging
+import sys
+import json
+
+import configparser
+import requests
+import subprocess
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+sh = logging.StreamHandler(sys.stdout)
+logger.addHandler(sh)
+
+state_persist: bool
+state_delay: int
+
+api_read_failure = False
+try:
+  response = requests.get('http://localhost:5001/settings/persist_logs', timeout=10)
+  state = response.json()
+
+  state_persist = state["persist_logs"]
+  state_delay = state["auto_off_delay"]
+except KeyError as exc:
+  logger.exception("Unable to read log persistence state through the api, attempting to read directly...")
+  api_read_failure = True
+except Exception as exc:
+  logger.exception(f"increment_auto_off.py has failed with the following exception:\n{exc}")
+  api_read_failure = True
+
+
+if api_read_failure:
+  try:
+    # If the api is inaccessible, assume it is down for whatever reason and extrapolate that there is no race condition to reading it directly
+    journalconf = configparser.ConfigParser(strict=False, allow_no_value=True)
+    journalconf.read('/etc/systemd/journald.conf')
+
+    logconf = configparser.ConfigParser(strict=False, allow_no_value=True)
+    logconf.read('/var/log/logging.ini')
+
+    persist = journalconf.get("Journal", "Storage", fallback="volatile")
+    state_persist = True if persist not in ("volatile", "persistent") else persist == "persistent"  # If value is somehow invalid, assume it's True just in case
```

Comment:
I assume that persistence is on in case you somehow corrupt your journald.conf file during a catastrophic issue, that way it remains on for troubleshooting purposes

I do not see a reality where you have that file set to an invalid option and meant for it to be off. If that is the case though, simply wait 14 days by default and you'll be fine.